src/custom/state/orders/updaters/UnfillableOrdersUpdater.ts

```diff
@@ -0,0 +1,147 @@
+import { useCallback, useEffect, useRef } from 'react'
+import { Price } from '@uniswap/sdk-core'
+
+import { useActiveWeb3React } from 'hooks/web3'
+
+import { usePendingOrders, useSetIsOrderUnfillable } from 'state/orders/hooks'
+import { Order } from 'state/orders/actions'
+import { OUT_OF_MARKET_PRICE_DELTA_PERCENTAGE, PENDING_ORDERS_PRICE_CHECK_POLL_INTERVAL } from 'state/orders/consts'
+
+import { SupportedChainId as ChainId } from 'constants/chains'
+import { ONE_HUNDRED_PERCENT } from 'constants/index'
+
+import { getBestPrice, PriceInformation } from 'utils/price'
+
+/**
+ * Thin wrapper around `getBestPrice` that builds the params and returns null on failure
+ *
+ * @param chainId
+ * @param order
+ */
+async function _getOrderPrice(chainId: ChainId, order: Order) {
+  let amount, baseToken, quoteToken
+
+  if (order.kind === 'sell') {
+    amount = order.sellAmount.toString()
+    baseToken = order.sellToken
+    quoteToken = order.buyToken
+  } else {
+    amount = order.buyAmount.toString()
+    baseToken = order.buyToken
+    quoteToken = order.sellToken
+  }
+
+  const quoteParams = {
+    chainId,
+    amount,
+    kind: order.kind,
+    baseToken,
+    quoteToken,
+    fromDecimals: order.inputToken.decimals,
+    toDecimals: order.outputToken.decimals,
+  }
+
+  try {
+    return await getBestPrice(quoteParams)
+  } catch (e) {
+    return null
+  }
+}
+
+/**
+ * Based on the order and current price, returns `true` if order is out of the market.
+ * Out of the market means the price difference between original and current to be positive
+ * and greater than OUT_OF_MARKET_PRICE_DELTA_PERCENTAGE.
+ * Negative difference is good for the user.
+ * We allow for it to be up to OUT_OF_MARKET_PRICE_DELTA_PERCENTAGE worse to account for
+ * small price changes
+ *
+ * @param order
+ * @param price
+ */
+function isOrderUnfillable(order: Order, price: Required<PriceInformation>): boolean {
```

Comment:
I like how you implemented all, you break things in small functions. 

Would it be worth it to do in another PR some test for this nice function, since it's a pure function and key for the logic in question?

Reply:
Yep, will do.

Reply:
https://github.com/gnosis/cowswap/pull/1260